test_programs/compile_success_empty/comptime_function_definition/src/main.nr

```diff
@@ -0,0 +1,31 @@
+use std::meta::type_of;
+
+struct Foo { x: Field }
+
+#[function_attr]
+fn foo(w: i32, y: Field, Foo { x }: Foo, mut a: bool) -> i32 {
+    1
+}
+
+comptime fn function_attr(f: FunctionDefinition) {
```

Comment:
Does this also work on functions defined within type or trait impl's?

Reply:
Do you mean putting an attribute over a function that's defined in a type impl or trait impl?

Reply:
@asterite Yes, exactly

Reply:
It seems that doesn't work. That said, I think that's functionality that already existed, here it's just about adding some `FunctionDefinition` methods (I didn't know how to get a `FunctionDefinition` without using an attribute... I think in the future we might be able to get them via `Module` but that's currently not implemented)